lab_4_retrieval_w_clustering/main.py

```diff
@@ -81,6 +104,16 @@ def _calculate_bm25(self, tokenized_document: list[str]) -> Vector:
         Returns:
             Vector: BM25 vector for document.
         """
+        if not tokenized_document or not isinstance(tokenized_document,list) or not\
+                all(isinstance(paragraph,str) for paragraph in tokenized_document):
+            raise ValueError
+        vector_list = [0.0 * i for i in range(len(self._vocabulary))]
```

Comment:
try not use list comprehension, how you can do it?

Reply:
I don't know, like that?

Reply:
yes!